pkgs/applications/video/pipe-viewer/default.nix

```diff
@@ -0,0 +1,95 @@
+{ buildPerlModule, lib, fetchFromGitHub, makeWrapper, wrapGAppsHook
+, withGtk3 ? false
+, ffmpeg
+, gtk3
+, wget
+, xdg-utils
+, youtube-dl
+# Perl dependencies
+, AnyURIEscape
+, DataDump
+, Encode
+, FilePath
+, FileShareDir
+, GetoptLong
+, Gtk3
+, HTTPMessage
+, JSON
+, JSONXS
+, LWPProtocolHttps
+, LWPUserAgentCached
+, Memoize
+, PathTools
+, ScalarListUtils
+, TermReadLineGnu
+, TestPod
+, TextParsewords
+, UnicodeLineBreak
```

Comment:
Maybe we can use a `perl.withPackages()` environment for this?  Is there a requirement for pipe-viewer to require these Perl packages directly on the user environment?

Reply:
@zakame, I've done that, except for `Gtk3` because it fails `wrapGAppsHook`.